### PR TITLE
Marketplace: Add Reviews Rating component to the reviews form

### PR DESCRIPTION
**Pull request by epeicher**
Related to #https://github.com/Automattic/dotcom-forge/issues/4661

## Proposed Changes

* Add ReviewsRatingsStars component to Calypso components. This component has been copied and updated from [WooCommerce component](https://github.com/Automattic/woocommerce.com/blob/a1f6b43a44b4f8c8a244586ad1ee371c82aaee8a/themes/woo/components/reviews/ratings-stars/ratings-stars.js)
* Use that component in the ReviewsModal to introduce the ratings
* Create a ReviewsSummary component that contains the reviews area in the Theme details screen

|Before | After|
|-------|------|
|      CleanShot 2023-12-05 at 19 46 56@2x   |       CleanShot 2023-12-05 at 19 46 16@2x            |

## Testing Instructions

* With the flag `marketplace-add-review` enabled
* Navigate to a theme details page, e.g. `/themes/ hevor/:siteId` selecting any site
* Click on the Add Review button
* Make sure you see the ReviewsRatingsStars component 
* Select a rating choosing the number of stars
* Make sure the component allows to select different ratings and updates accordingly
* Click on Submit
* Make sure the network call `https://public-api.wordpress.com/wp/v2/sites/marketplace.wordpress.com/comments` contains the selected rating and it completes as expected. Please note that if there is a validation error in the backend, an error will be returned.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/components/reviews-rating-stars/reviews-ratings-stars.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+			break;
+		case 'medium-large':
+			classNames.push( 'reviews-ratings-stars__star-bar--medium-large' );
+			break;
+	}
+
+	// Show numberOfReviews if the prop was passed
+	let numberOfReviewsText;
+	if ( numberOfReviews ) {
+		numberOfReviewsText = translate( '%(numberOfReviews) review', '%(numberOfReviews) reviews', {
```
Comment (WBerredo) [inline client/components/reviews-rating-stars/reviews-ratings-stars.tsx]: Is this translation working? I expect it to need something like `ranslate( '%(numberOfReviews)d review',...` with the `d` or `s` after the parenthesis?

File: client/components/reviews-rating-stars/reviews-ratings-stars.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+			onSelectRating( index );
+		}
+	};
+
+	useEffect( () => {
+		setRatingState( rating );
+		setHoverRating( rating );
+	}, [ rating ] );
+
+	const classNames = [ 'reviews-ratings-stars__star-bar' ];
```
Comment (WBerredo) [inline client/components/reviews-rating-stars/reviews-ratings-stars.tsx:55]: Nit: Maybe the `className` prop is also worth adding here so we can inject different classes if needed.

File: client/components/reviews-rating-stars/reviews-ratings-stars.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+						</div>
+					) }
+					<div className="reviews-ratings-stars__stars" aria-hidden="false">
+						{ Array( MAX_RATING )
+							.fill( null )
+							.map( ( _, index ) => {
+								const value = index + 1;
+								const ratingStarsScreenReaderText = translate(
+									'Rate product %(value) star',
+									'Rate product %(value) stars',
```
Comment (WBerredo) [inline client/components/reviews-rating-stars/reviews-ratings-stars.tsx]: Same as above about those variables

File: client/components/reviews-rating-stars/reviews-ratings-stars.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+			onMouseEnter={ onMouseEnter }
+			onMouseLeave={ onMouseLeave }
+		>
+			{ simpleView ? (
+				<Rating rating={ ratingNormalized } />
+			) : (
+				<>
+					{ averageRating && (
+						<div className="reviews-ratings-stars__average">
+							<span className="screen-reader-text">{ translate( 'Average rating ' ) }</span>
```
Comment (WBerredo) [inline client/components/reviews-rating-stars/reviews-ratings-stars.tsx]: Nit: Is this space at the end of the string needed?

File: client/components/reviews-rating-stars/reviews-ratings-stars.tsx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+					</div>
+					{ numberOfReviewsText && (
+						<div className="reviews-ratings-stars__rating-count">{ numberOfReviewsText }</div>
+					) }
+					{ selectedRatingText && (
+						<div className="reviews-ratings-stars__selected-rating">{ selectedRatingText }</div>
+					) }
+				</>
+			) }
+			{ children && children }
```
Comment (WBerredo) [inline client/components/reviews-rating-stars/reviews-ratings-stars.tsx]: Any reason to not put just `{children}`?

**Review by WBerredo — APPROVED** [review]
Added some non-blocking comments below, but the behavior LGTM!

**Comment by JulianBustamante** [comment]
I tested it out and it worked as expected.

Screenshot 2023-12-12 at 10 08 55 PM

When I click on a star, a line surrounding the image shows up. The surrounding line can be seen in the attached image. It is possible to avoid displaying that line?

Also, we need to refine the message returned when the comment is empty.

**Review by JulianBustamante — APPROVED** [review]

File: client/components/reviews-rating-stars/reviews-ratings-stars.tsx (hunk 1 above)
Comment (epeicher) [reply client/components/reviews-rating-stars/reviews-ratings-stars.tsx]: Good catch, this is not used at the moment by the consumer but it has the bug you mention that I will fix

File: client/components/reviews-rating-stars/reviews-ratings-stars.tsx (hunk 4 above)
Comment (epeicher) [reply client/components/reviews-rating-stars/reviews-ratings-stars.tsx]: Nope, removed e60a96be49119ec8e70e34cfe70112f86a0c0064

File: client/components/reviews-rating-stars/reviews-ratings-stars.tsx (hunk 5 above)
Comment (epeicher) [reply client/components/reviews-rating-stars/reviews-ratings-stars.tsx]: Not really, it comes from the original implementation and I think it is an old pattern to check if children exists, but I don't think is required and it hurts readability so I have removed it in e60a96be49119ec8e70e34cfe70112f86a0c0064

File: client/components/reviews-rating-stars/reviews-ratings-stars.tsx (hunk 3 above)
Comment (epeicher) [reply client/components/reviews-rating-stars/reviews-ratings-stars.tsx]: Fixed here e60a96be49119ec8e70e34cfe70112f86a0c0064.

File: client/components/reviews-rating-stars/reviews-ratings-stars.tsx (hunk 1 above)
Comment (epeicher) [reply client/components/reviews-rating-stars/reviews-ratings-stars.tsx]: Fixed here e60a96be49119ec8e70e34cfe70112f86a0c0064

**Comment by epeicher** [comment]
Thanks for your review @JulianBustamante !

> When I click on a star, a line surrounding the image shows up. The surrounding line can be seen in the attached image. It is possible to avoid displaying that line?

We can certainly remove the outline, but it is not recommended to remove it for people navigating with a keyboard https://www.a11yproject.com/posts/never-remove-css-outlines/, please see gif below
CleanShot 2023-12-13 at 11 06 20


> Also, we need to refine the message returned when the comment is empty.

I agree, we can tackle all the validation messages when refining the component. They are the ones returned from the backend

File: client/components/reviews-rating-stars/reviews-ratings-stars.tsx (hunk 2 above)
Comment (epeicher) [reply client/components/reviews-rating-stars/reviews-ratings-stars.tsx:55]: That's a great pattern for extensibility. We don't use extra classes at the moment so I will leave it for potential future extensions

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10265673

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @epeicher for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10265673).

**Pull request merged**